src/neptune/common/warnings.py

```diff
@@ -55,16 +56,19 @@ def get_user_code_stack_level():
 
 
 def warn_once(message: str, *, exception: type(Exception) = None):
-    if exception is None:
-        exception = NeptuneDeprecationWarning
-
-    if message not in warned_once:
-        warnings.warn(
-            message=message,
-            category=exception,
-            stacklevel=get_user_code_stack_level(),
-        )
-        warned_once.add(message)
+    if len(warned_once) < 1_000:
+        if exception is None:
+            exception = NeptuneDeprecationWarning
+
+        message_hash = hashlib.sha256(message.encode("utf-8")).hexdigest()
```

Comment:
You can simply use the `hash()` builtin instead of sha

Reply:
done, in first version I used to python hash.